# Práctica 4

Realizar los ejercicios de las notas (en papel excepto en los ejercicios que explícitamente se mencione se hagan en computadora en la nota correspondiente):

+ Propedeutico/R/clases/2_probabilidad/1_elementos_de_probabilidad.ipynb
+ Propedeutico/R/clases/2_probabilidad/2_propiedades_de_la_probabilidad.ipynb
+ Propedeutico/R/clases/2_probabilidad/3_metodos_de_conteo.ipynb
+ Propedeutico/R/clases/2_probabilidad/4_probabilidad_condicional.ipynb
+ Propedeutico/R/clases/2_probabilidad/5_independencia.ipynb
+ Propedeutico/R/clases/2_probabilidad/6_teorema_de_Bayes.ipynb
+ Propedeutico/R/clases/2_probabilidad/7_variables_aleatorias.ipynb

Leer, ejecutar y proponer sus propios ejemplos de uso de lenguaje para los ejemplos (en computadora):

+ Propedeutico/R/clases/1_introduccion/2_core_R.ipynb
+ Propedeutico/R/clases/1_introduccion/3_ggplot2.ipynb

In [1]:
options(repr.plot.width=4, repr.plot.height=4) #esta línea sólo se ejecuta para jupyterlab con R

In [2]:
install.packages('prob')
library(prob)

Updating HTML index of packages in '.Library'
Making 'packages.html' ... done
Loading required package: combinat

Attaching package: ‘combinat’

The following object is masked from ‘package:utils’:

    combn

Loading required package: fAsianOptions
Loading required package: timeDate
Loading required package: timeSeries
Loading required package: fBasics
Loading required package: fOptions

Attaching package: ‘prob’

The following objects are masked from ‘package:base’:

    intersect, setdiff, union



## Parte 1: 1_elementos_de_probabilidad

### Ejemplos

### ¿Cómo representar en R un espacio de probabilidad?

Una opción para responder esta pregunta es considerar un objeto, `S`, que represente los *outcomes* o resultados del experimento y un vector de probabilidades, `probs`, con entradas que correspondan a cada outcome en `S`.

Además en el paquete *prob* se tiene una función `probspace` que tiene por argumentos $x$ que es un espacio muestral de los outcomes y $probs$ es un vector del mismo tamaño que el número de outcomes en $x$.

#### 1) Lanzamiento de un dado justo

In [3]:
outcomes <- rolldie(1)
p <- rep(1/6, times = 6)
probspace(outcomes, probs = p)
#y es equivalente sólo haber ejecutado: 
#probspace(1:6, probs = p) o bien probspace(1:6) o rolldie(1, makespace = TRUE)

X1 probs    
1 1  0.1666667
2 2  0.1666667
3 3  0.1666667
4 4  0.1666667
5 5  0.1666667
6 6  0.1666667

#### 2) Lanzamiento de una moneda cargada

Supóngase que $P(\{\text{sol}\}) = .7$ y $P(\{\text{águila}\}) = .3$ entonces:

In [4]:
probspace(tosscoin(1), probs = c(0.70, 0.30))

toss1 probs
1 H     0.7  
2 T     0.3

**Ejercicio:** ¿cómo calcular la probabilidad anterior con la función `urnsamples`?

En el paquete *prob* se tiene la función *urnsamples* la cual tiene argumentos $x, size, replace, ordered$. El argumento $x$ representa la urna de la cual se realizará el muestreo, $size$ representa el tamaño de la muestra, $replace$ y $ordered$ son argumentos lógicos y especifican cómo se debe realizar el muestreo.

One sample space function that does NOT have a makespace option is the urnsamples() function. This was intentional. The reason is that under the varied sampling assumptions the outcomes in the respective sample spaces are NOT, in general, equally likely. It is important for the user to carefully consider the experiment to decide whether or not the outcomes are equally likely, and then use probspace() to assign the model.
https://cran.r-project.org/web/packages/prob/vignettes/prob.pdf

Documentación de Probspace:
https://www.rdocumentation.org/packages/prob/versions/1.0-1/topics/probspace

Documentación de Urnsamples:
https://www.rdocumentation.org/packages/prob/versions/1.0-1/topics/urnsamples

In [5]:
#inicializando la "urna", que serían los valores de la moneda
x <- c('Sol', 'Aguila')
muestra<-urnsamples(x, 1)
muestra

out   
1 Sol   
2 Aguila

In [6]:
probspace(muestra,probs = c(0.70, 0.30))

out    probs
1 Sol    0.7  
2 Aguila 0.3

## Parte 1: 2_propiedades_de_la_probabilidad

### Ejercicios (a mano)

Calcula las siguientes probabilidades suponiendo el modelo equiprobable y experimentos justos:

1) Durante 5 años el profesor X ha impartido el mismo curso. En total ha tenido 500 alumnos, de los cuales 80 se han dado de baja, 200 han reprobado y 220 han aprobado. Nos encontramos a uno de estos exalumnos. ¿Cuál es la probabilidad de que éste se haya dado de baja en el curso?

2) Una moneda se lanza al aire 300 veces y se obtienen 152 águilas y 148 soles. ¿Cuál es la probabilidad de sacar un águila?

3) El año pasado nacieron 1613 mujeres y 1531 hombres en la Delegación Coyoacán. Si se selecciona uno de estos bebes al azar, ¿cuál es la probabilidad de que sea niña? 

4) Se lanza un dado de seis caras al aire. Sean los eventos $E=${el resultado es un número par} y $F=${el resultado es un número primo}. Calcular la probabilidad de “el resultado es un número par o un número primo”.

5) ¿Cuál es la probabilidad de tener exactamente dos hijos varones en una familia de tres hijos?  ¿a lo más dos hijos varones?

6) Considera una baraja ¿cuál es la probabilidad de sacar un as? ¿de un corazón? ¿de un as de corazones? ¿de un as o de un corazón?

7) Un sondeo de opinión entre la población adulta acerca de si se desea una nueva línea del metro nos da los siguientes resultados:


|----|A favor|En contra|Total|
|:---:|:---:|:---:|:---:
|Mujeres|210|50|260|
|Hombres|180|60|240|
|Total|390|110|500|

Sea $F$ el evento “el encuestado es mujer” y $C$ el evento “está en contra de la nueva línea del Metro”. Calcular $P(F), P(C), P(F \cap C), P(F \cup C)$.

## Parte 1: 3_metodos_de_conteo

### Ejercicios (a mano)

Resolver lo siguiente:

1) Tenemos 5 ingredientes de los cuales podemos elegir para pedir nuestra pizza: peperoni, salchicha, anchoas, huitlacoche y aceitunas. ¿Cuántas pizzas distintas podemos formar?

2) Para construir una computadora podemos elegir entre las opciones que están en una tienda: dos procesadores, tres sistemas operativos, cuatro tamaños distintos de memoria, 4 discos duros de distinto tamaño y 10 opciones de monitor. ¿Cuántas opciones de computadoras podemos construir?

3) ¿Cuántas formas distintas se pueden construir números de tres cifras sin repetir ningún número?

4) Cuatro personas ordenan diferentes bebidas en un bar. El cantinero sirve las bebidas pero por falta de atención olvida completamente quién ordenó qué. ¿Cuál es la probabilidad de que el cantinero coloque las bebidas correctamente? 

5) Rentas $5$ películas para ver: $3$ hoy y $2$ mañana por la noche pero tu amiga Mariana te pide prestadas $2$ películas. ¿De cuántas formas distintas puede Mariana elegir las $2$ películas?

6) ¿Cuántas manos de poker contienen (exactamente) una tercia? ---> (excluye par, esto es, excluye el llamado full)

## Parte 1: 4_probabilidad_condicional

## Propiedades y reglas

Las probabilidades condicionales se comportan como probabilidades sin condicionar:

Para cualquier evento $E$ fijo con $P(E)>0$ se tiene:

1) $P(F|E) \geq 0$, para todos los eventos $F \in S$.

2) $P(S|E) = 1$.

3) Si los eventos $F_1, F_2, F_3, \dots $ son ajenos entonces:

$P(\displaystyle \cup_{i=1}^n F_i | E ) =  \sum_{i=1}^nP(F_i | E)$ para toda $n$ y aún más: $P(\displaystyle \cup_{i=1}^\infty F_i| E ) = \sum_{i=1}^\infty P(F_i|E)$

Y se tienen las siguientes propiedades:

1)  $P(F^c | E) = 1-P(F|E)$.

2) Si $F \subset G$ entonces $P(F|E) \leq P(G|E)$.

3) $P(F \cup G|E) = P(F|E) + P(G|E) - P(F \cap G|E)$ 

4) La **regla de la multiplicación** para cualquier par de eventos $E, F$: $P(E \cap F) = P(E|F)P(F) = P(F|E)P(E)$ y en general: $P(\displaystyle \cap_{i=1}^n E_i) = P(E_1)P(E_2|E_1)P(E_3|E_1 \cap E_2) \cdots P(E_n |\displaystyle \cap_{i=1}^{n-1} E_i)$

### Ejemplos de la regla de la multiplicación


1)**Problema:** calcular la probabilidad de extraer $2$ aces de una baraja.

**Solución:**

Si $E=${extraer un as} y $F=${extraer un as} entonces $P(E\cap F)=P(E|F)P(F) = \frac{3}{51} \cdot \frac{4}{52}  \approx 0.00452$

#### En R ...

In [7]:
L <- cards() #Es un dataframe
head(L)

rank suit
1 2    Club
2 3    Club
3 4    Club
4 5    Club
5 6    Club
6 7    Club

In [8]:
M <- urnsamples(L, size = 2, replace=F, ordered = F) #extraer dos canicas
                                                     #sin reemplazo y sin orden
N <- probspace(M) #modelo equiprobable

$M$ es una lista de una dimensión que contiene todas las combinaciones posibles de pares de renglones del data frame $L$. La cantidad de combinaciones de pares de renglones es $\binom{52}{2}:$

In [9]:
NCOL(M)

[1] 1

In [10]:
NROW(M)

[1] 1326

In [11]:
choose(52,2)

[1] 1326

In [12]:
head(M)

[[1]]
  rank suit
1    2 Club
2    3 Club

[[2]]
  rank suit
1    2 Club
3    4 Club

[[3]]
  rank suit
1    2 Club
4    5 Club

[[4]]
  rank suit
1    2 Club
5    6 Club

[[5]]
  rank suit
1    2 Club
6    7 Club

[[6]]
  rank suit
1    2 Club
7    8 Club

$N$ es una lista de dimensión igual a dos, se utiliza `attributes` para revisar sus `names`: 

In [13]:
print(attributes(N))

$names
[1] "outcomes" "probs"   

$class
[1] "ps"   "list"



In [14]:
print(attributes(N)$names) #dos dimensiones

[1] "outcomes" "probs"   


En la primer dimensión tiene $\binom{52}{2} = 1326$ data frames y cada data frame tiene 2 renglones:

In [15]:
length(N$outcomes)

[1] 1326

In [16]:
N$outcomes[1]

[[1]]
  rank suit
1    2 Club
2    3 Club

In [17]:
N$outcomes[100]

[[1]]
   rank  suit
2     3  Club
51    K Spade

Cada renglón representa el par de cartas elegidas.

La segunda dimensión tiene $\binom{52}{2} = 1326$ probabilidades:

In [18]:
length(N$probs)

[1] 1326

cada probabilidad igual a $\frac{1}{\binom{52}{2}}$ que es la probabilidad de extraer dos cartas cualesquiera.

In [19]:
N$probs[1]

[1] 0.0007541478

In [20]:
N$probs[100]

[1] 0.0007541478

In [21]:
1/choose(52,2)

[1] 0.0007541478

Y la probabilidad deseada: $P(E\cap F) = P(${extraer dos aces}$)$ es:

In [22]:
Prob(N, all(rank == 'A')) #Prob asume modelo equiprobable
                          # el evento es all(rank == 'A') y como son dos cartas extraídas, 
                          #entonces ambas deben ser iguales a A

[1] 0.004524887

2) **Problema:** considérese una urna con $10$ canicas: $7$ son rojas y $3$ son verdes y se seleccionan $3$ canicas. Sea $E=${primer canica es roja}, $F=${segunda canica es roja} y $G=${tercera canica es roja}. Calcular $P(E \cap F \cap G)$.

**Solución:**

$P(E\cap F\cap G) = \frac{\binom{7}{3}}{\binom{10}{3}}$

In [23]:
choose(7,3)/choose(10,3)

[1] 0.2916667

#### En R ...

In [24]:
L <- rep(c("red","green"), times = c(7,3)) #urna, es un vector
print(L)

 [1] "red"   "red"   "red"   "red"   "red"   "red"   "red"   "green" "green"
[10] "green"


In [25]:
M <- urnsamples(L, size = 3, replace = FALSE, ordered = TRUE) #data frame, espacio de resultados

In [26]:
NROW(M)

[1] 720

In [27]:
NCOL(M)

[1] 3

In [28]:
M[35:40,]

X1    X2    X3   
35 red   green red  
36 red   red   green
37 red   red   green
38 red   green red  
39 green red   red  
40 green red   red

In [29]:
N <- probspace(M) #a cada renglón de M (outcomes) se le asocia un vector de probabilidades
                  #con lo que se tiene un data frame: N, que es un espacio de probabilidad
                  #probspace usa modelo equiprobable

In [30]:
NROW(N)

[1] 720

In [31]:
NCOL(N)

[1] 4

In [32]:
N

X1    X2    X3    probs      
1   red   red   red   0.001388889
2   red   red   red   0.001388889
3   red   red   red   0.001388889
4   red   red   red   0.001388889
5   red   red   red   0.001388889
6   red   red   red   0.001388889
7   red   red   red   0.001388889
8   red   red   red   0.001388889
9   red   red   red   0.001388889
10  red   red   red   0.001388889
11  red   red   red   0.001388889
12  red   red   red   0.001388889
13  red   red   red   0.001388889
14  red   red   red   0.001388889
15  red   red   red   0.001388889
16  red   red   red   0.001388889
17  red   red   red   0.001388889
18  red   red   red   0.001388889
19  red   red   red   0.001388889
20  red   red   red   0.001388889
21  red   red   red   0.001388889
22  red   red   red   0.001388889
23  red   red   red   0.001388889
24  red   red   red   0.001388889
25  red   red   red   0.001388889
26  red   red   red   0.001388889
27  red   red   red   0.001388889
28  red   red   red   0.001388889
29  red   red   red   0.001388889
30  red   red   red   0.001388889
⋮   ⋮     ⋮     ⋮     ⋮          
691 red   green green 0.001388889
692 red   green green 0.001388889
693 green red   green 0.001388889
694 green green red   0.001388889
695 green green red   0.001388889
696 green red   green 0.001388889
697 red   green green 0.001388889
698 red   green green 0.001388889
699 green red   green 0.001388889
700 green green red   0.001388889
701 green green red   0.001388889
702 green red   green 0.001388889
703 red   green green 0.001388889
704 red   green green 0.001388889
705 green red   green 0.001388889
706 green green red   0.001388889
707 green green red   0.001388889
708 green red   green 0.001388889
709 red   green green 0.001388889
710 red   green green 0.001388889
711 green red   green 0.001388889
712 green green red   0.001388889
713 green green red   0.001388889
714 green red   green 0.001388889
715 green green green 0.001388889
716 green green green 0.001388889
717 green green green 0.001388889
718 green green green 0.001388889
719 green green green 0.001388889
720 green green green 0.001388889

In [33]:
Prob(N, isrep(N, vals='red', nrep=3))# equivalente hacer: 
                                     #Prob(N, X1 =='red'& X2 =='red' & X3 =='red')
                                     #y se calculó: sum(N[with(N,isrep(N, 'red', 3)),]$probs)

[1] 0.2916667

`isrep` es una función del paquete `prob`

**(En computadora) Ejercicio: calcular la probabilidad de:**

* extraer dos canicas rojas

* extraer roja, verde, roja (en este orden)

* extraer roja, verde, roja (no importa el orden)

#### Ejercicio 2.1: Extraer dos canicas rojas

Documentación de la función prob: https://cran.r-project.org/web/packages/prob/prob.pdf

extraer dos canicas rojas:

si son dos canicas rojas, independientemente de su posición, sería:

In [46]:
Prob(N, isrep(N, vals='red', nrep=2))

[1] 0.525

Si son 2 canicas rojas en las primeras 2 posiciones:

In [47]:
Prob(N, X1 =='red'& X2 =='red')

[1] 0.4666667

extraer roja, verde, roja (en este orden):

In [48]:
Prob(N, X1 =='red'& X2 =='green' & X3 =='red')

[1] 0.175

extraer roja, verde, roja (no importa el orden):<br>
Esto equivale a 2 rojas y 1 verde o a la suma de la verde en todas las posiciones:

In [49]:
Prob(N, X1 =='green'& X2 =='red' & X3 =='red')+Prob(N, X1 =='red'& X2 =='green' & X3 =='red')+Prob(N, X1 =='red'& X2 =='red' & X3 =='green')

[1] 0.525

Otra forma de escribir lo anterior es:

In [50]:
Prob(N, isrep(N, vals='red', nrep=2))

[1] 0.525

### Ejercicios (a mano)

1) Supongamos que en una muestra de 100 personas, 65 de ellas son vacunadas contra la influenza esta temporada de invierno. Cinco de las personas vacunadas contraen la enfermedad. De las 35 personas que no son vacunadas 25 caen enfermas. La siguiente tabla resume esta información:

|----|enfermo|no enfermo|Total|
|:---:|:---:|:---:|:---:|
|vacunados|5|60|65|
|no vacunados|25|10|35|
|Total|30|70|100

Calcular:

a) $P(${vacunado}$)$, $P(${no vacunado}$)$, $P(${enfermo|vacunado}$)$, $P(${no enfermo|vacunado}$)$, $P(${enfermo|no vacunado}$)$, $P(${no enfermo|no vacunado}$)$ y $P(${vacunado y no enfermo}$)$. 

b)$P(${no vacunado y no enfermo}$)$, $P(${no enfermo}$)$

c)$P(${vacunado} | {no enfermo}$)$.



## Parte 1: 5_independencia

### Ejercicios (a mano)

1) Calcular la probabilidad de sacar un as y corazón en una baraja

2) Se lanzan dos dados, uno verde y otro rojo. Considérense los eventos $E=${los números suman 5}, $F=${el número en el dado verde es impar}. ¿$E$ y $F$ son independientes?.

3) Supóngase que nuestra elección de postre cuando salimos a comer es independiente del platillo principal que pedimos. La mitad de las veces que vamos a comer al restaurante "L Intervallo" pedimos pasta con camarones y la mitad de las veces pedimos *pannacotta* de postre. ¿Cuál es la probabilidad de ordenar pasta con camarones y *pannacotta* dado que comemos en "L Intervallo"?

4) Sea E el evento de que el número $111111$ sea ganador en el sorteo Melate (la lotería Melate consiste en elegir una combinación de seis números diferentes entre el $1$ y el $56$, sin orden y sin reemplazo) y $F$ el evento de que el número $555555$ sea el ganador.

a) Determina si estos eventos son mutuamente excluyentes.

b) Determina si estos eventos son independientes.

## Parte 1: 6_teorema_de_Bayes

### Ejercicios (a mano)

1)El siguiente es un mecanismo conocido para elegir a una persona de un grupo pequeño. Digamos que en un grupo de 4 personas quiere seleccionarse a una de ellas para que realice alguna actividad, digamos ir por cervezas a la hielera. Para este efecto, se toman 4 palillos y uno de ellos se recorta; así, se tienen tres palillos largos y uno corto. Las personas toman turnos tomando palillos, ignorando cuál es el corto, y el que saca el palillo corto es el elegido para ir por las cervezas. ¿Tiene alguna ventaja escoger primero, segundo, tercero o cuarto?

2)Considera el mismo escenario que el ejemplo $3$ pero en lugar de que el sospechoso sea un individuo albino ahora se tiene:

a) El sospechoso del crimen es un individuo pelirrojo y la probabilidad de ser pelirrojo es del $2 \%$. Encontrar $P(pelirrojo | Tp),$ en donde $Tp$ representa: “el testigo identifica a una persona como pelirroja”.

b) El sospechoso es un individuo de pelo rubio y la probabilidad de tener el pelo rubio es del $15 \%$. Encontrar $P(rubio | Tr)$, en donde $Tr$ representa: “el testigo identifica a una persona como rubia”.

## Parte 1: 7_variables_aleatorias

### Ejercicios (a mano)

1) Considérese lanzar dos monedas. Sea $X$ la variable aleatoria que cuenta el número de águilas que aparecen. Calcula la probabilidad de $P(X=0), P(X=1), P(X=2)$.

2) Considérese el sorteo de Melate (elegir una combinación de seis números diferentes entre el $1$ y el $56$ sin orden y sin reemplazo) "simplificado" con una bolsa acumulada de $N$ y la variable aleatoria $X$ que toma dos valores: $N$ si se elige la combinación ganadora y $0$ en cualquier otro caso. ¿Cuál es la distribución de probabilidad de $X$?

(es simplificado pues en realidad la bolsa acumulada se reparte entre el número de personas ganadoras y varios premios menores, según el número de cifras acertadas entre dos y seis.)

3) Calcular el valor esperado de los ejemplos $1)$ y $3)$ de la sección "Distribución de probabilidad de una variable aleatoria" y responder: ¿Si nos ofrecieran jugar uno de estos dos juegos cuál deberíamos elegir?. Nótese que ninguno de los premios o pérdidas es igual al valor esperado, éste simplemente nos da una idea del promedio de los premios y pérdidas que resultarían si el juego se repitiese un gran número de veces.


4)  Los juegos de azar en los casinos pueden representarse por medio de variables aleatorias. Consideremos una versión simplificada de una máquina de palanca: existen dos figuras, digamos un cuadrado y un triángulo, que pueden aparecer en tres columnas de una cinta circular. El jugador baja la palanca, las cintas giran y al detenerse cada columna muestra un cuadrado o un triángulo. Las $8$ posibilidades equiprobables son: 

$$\begin{array}{ccc} \blacksquare &\blacksquare& \blacksquare \\ \blacksquare & \blacksquare& \blacktriangle \\ \blacksquare & \blacktriangle & \blacksquare \\ \blacksquare & \blacktriangle & \blacktriangle \\ \blacktriangle & \blacksquare & \blacksquare \\ \blacktriangle & \blacksquare & \blacktriangle \\ \blacktriangle & \blacktriangle & \blacksquare \\ \blacktriangle & \blacktriangle & \blacktriangle \end{array}$$

Supongamos que para jugar se introduce una ficha de $\$300$ y los premios asociados son: $\$500$ si salen tres figuras iguales y $\$0$ de cualquier otra forma. Sea $X$ la variable aleatoria que mide la ganancia o pérdida de jugar en la máquina. Calcular $E(X)$.

### Ejercicios (en computadora)

Los siguientes ejercicios realizarlos en `R` y con `ggplot2`:

5) Encontrar los histogramas de probabilidad para las distribuciones asociadas a las siguientes variables aleatorias:

a) El número de águilas que aparecen cuando lanzamos tres monedas al aire.
    
b) La suma de los números de las caras cuando se lanzan dos dados.

6) Se venden 8000 boletos para una rifa de $\$5000.00$ y cada boleto cuesta $\$2.00$.

a) Encontrar la ganancia esperada del comprador de un boleto.
 
b) Hacer la gráfica de la ganancia esperada que tiene una compradora en términos de un número de boletos $n$.
 
c) ¿Cuál debería de ser el premio mínimo para que se pudiese garantizar “salir a mano” al comprar todos los boletos.

## Parte 2: 2_core_R

## Parte 2: 3_ggplot2